Application/EdFi.Ods.Tests/EdFi.Ods.Api/Providers/Aes256SymmetricStringEncryptionProviderTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.Ods.Api.Providers;
+using Shouldly;
+
+namespace EdFi.Ods.Tests.EdFi.Ods.Api.Providers;
+
+using System;
+using NUnit.Framework;
+
+public class Aes256SymmetricStringEncryptionProviderTests
+{
+    private Aes256SymmetricStringEncryptionProvider _encryptionProvider;
+    private byte[] _staticKey;
+    [SetUp]
+    public void SetUp()
+    {
+        _encryptionProvider = new Aes256SymmetricStringEncryptionProvider();
+        _staticKey = Convert.FromBase64String("t3OD+5YCKzCdGSaWBAvkQ9+d9R4ZbZnC9YG8L/ar8vA=");
+    }
+
+    [Test]
+    public void When_encrypting_then_decrypting_a_string()
+    {
+        // Arrange
+        var plainText = "Hello World";
+
+        // Act
+        string encryptedString = _encryptionProvider.Encrypt(plainText, _staticKey);
+        bool decryptedSuccessfully = _encryptionProvider.TryDecrypt(encryptedString, out string decryptionResult, _staticKey);
+
+        // Assert
+        decryptedSuccessfully.ShouldBeTrue();
+        decryptionResult.ShouldBe(plainText);
+    }
+    
+    [Test]
+    public void When_decrypting_a_string_using_the_correct_key()
+    {
+        // Arrange
+        var cipherText = "Gx5FXSHSelzRF2quaCOaNw==.RlK3dR28TP/dbslPezxa4w==.D7VcBfHXG0iPkPR36QDvv9W7SGiGWUUe7WBu5neV/fo=";
+
+        // Act
+        bool decryptionSuccessful = _encryptionProvider.TryDecrypt(cipherText, out string output, _staticKey);
+
+        // Assert
+        decryptionSuccessful.ShouldBeTrue();
+        output.ShouldBe("Hello World");
+    }
+    
+    [Test]
+    public void When_hmac_signature_of_ciphertext_is_invalid()
+    {
+        // Arrange
+        var cipherText = "Gx5FXSHSelzRF2quaCOaNw==.RlK3dR28TP/dbslPezxa4w==.D7VcBfHXG0iPkPR36QDvv8W7SGiGWUUe7WBu5neV/fo=";

```

Comment:
I had to take this to kdiff to tell what the difference was in the ciphertext. Could you add a comment identifying the difference between this string and the good one in the previous test, and maybe also consider changing the leading `D` to an `X` instead of changing the `9` to an `8` in the middle of the string (which is quite a bit more subtle)? Another option would be to change the characters all to `X` for the entire length of the signature string. 

**Just need to make sure that it's testing the same condition -- that it base64 decodes the text without error and then fails on hmac signature validation.**

Basically, just try to make it a little easier to quickly understand what the difference is between the two strings.

Reply:
The difference in the base64 HMAC signature strings is changed to affect only the last non-padding character of the HMAC signature, and I added a comment explaining the difference. Additionally, this test now checks that the valid HMAC signature verifies before checking that the modified HMAC value causes a failure. Some other new comments also make the tests easier to follow.